src/Controller/CRUDController.php

```diff
@@ -212,10 +212,15 @@ public function batchActionDelete(ProxyQueryInterface $query)
      *
      * @return Response|RedirectResponse
      */
-    public function deleteAction($id)
+    public function deleteAction($id = null)
     {
         $request = $this->getRequest();
-        $id = $request->get($this->admin->getIdParameter());
+
+        if (null === $id) {
+            // Depending on the route declaration, the id can be a query parameter
+            $id = $request->get($this->admin->getIdParameter());
```

Comment:
Are you sure this does not work in both cases?

Reply:
I found weird the fact the $id wasn't used when he was provided.
But yeah, it works in both cases.

I think since we'll remove the `QueryStringBuilder` the code should be in the `4.0` version
```
    public function  someAction($id)
    {
        $request = $this->getRequest();
        $object = $this->admin->getObject($id);

        ...
    }
```

So I can actually do:
```
    public function  someAction($id = null) // NEXT_MAJOR: remove the default null value
    {
        $request = $this->getRequest();
        // NEXT_MAJOR: remove the next line
        $id = $request->get($this->admin->getIdParameter());
        $object = $this->admin->getObject($id);

        ...
    }
```

Reply:
Version 2 sounds cool 👍